internal/command/program_resolver_test.go

```diff
@@ -2,205 +2,264 @@ package command
 
 import (
 	"bytes"
-	"fmt"
 	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-type parsingFixture struct {
-	name   string
-	data   string
-	source []ProgramResolverSource
-	result *ProgramResolverResult
-	script string
-}
-
-func runProgramResolverFixtureTestCase(tc parsingFixture, mode ProgramResolverMode) func(t *testing.T) {
-	return func(t *testing.T) {
-		r := NewProgramResolver(mode, tc.source...)
-
-		if mode != ProgramResolverModeAuto && tc.result != nil {
-			tc.result.Prompt = ProgramResolverResolved
-			tc.result.Value = tc.result.OriginalValue
-		}
-
-		var script bytes.Buffer
-		result, err := r.Resolve(strings.NewReader(tc.data), &script)
-		assert.NoError(t, err)
-		assert.EqualValues(t, []byte(tc.script), script.Bytes())
-
-		var tcres []*ProgramResolverResult
-		if tc.result != nil {
-			tcres = append(tcres, tc.result)
-		}
-
-		assert.EqualValues(t, tcres, result)
-	}
-}
-
-func TestProgramResolverAutoAndSkip(t *testing.T) {
-	fixtures := []parsingFixture{
+func TestProgramResolverResolve(t *testing.T) {
+	testCases := []struct {
+		name            string
+		program         string
+		result          *ProgramResolverResult
+		modifiedProgram string
+	}{
 		{
-			name: "no value",
-			data: `export TEST_NO_VALUE`,
+			name:    "no value",
+			program: `export TEST_NO_VALUE`,
 			result: &ProgramResolverResult{
-				Name: "TEST_NO_VALUE", Prompt: ProgramResolverMessage,
+				ModifiedProgram: true,
+				Variables: []ProgramResolverVarResult{
+					{
+						Status: ProgramResolverStatusUnresolved,
```

Comment:
@sourishkrout There are a few changes like this one in these tests, which I hope led to more consistent behaviour. 

All variables are unresolved here and:

* If a variable is empty (`export A`, `export A=`, `export A=""`) it is marked as `ProgramResolverStatusUnresolved`.
* If there is any unquoted value, it is marked as `ProgramResolverStatusUnresolvedWithMessage`. 
* If there is any quoted value, it is marked as `ProgramResolverStatusUnresolvedWithPlaceholder`.

Reply:
That's fair. In any case, the extension will likely choose to prompt for `ProgramResolverStatusUnresolved` with a generic message. 👍